vertical-pod-autoscaler/pkg/recommender/main.go

```diff
@@ -86,7 +92,12 @@ func main() {
 	postProcessors := []routines.RecommendationPostProcessor{
 		&routines.CappingPostProcessor{},
 	}
-	recommender := routines.NewRecommender(config, *checkpointsGCInterval, useCheckpoints, *vpaObjectNamespace, *recommenderName, postProcessors)
+
+	var externalClientOptions *metrics2.ExternalClientOptions = nil
+	if *useExternalMetrics {
+		externalClientOptions = &metrics2.ExternalClientOptions{CpuMetric: *externalCpuMetric, MemoryMetric: *externalMemoryMetric, PodNamespaceLabel: *podNamespaceLabel, PodNameLabel: *podNameLabel, CtrNamespaceLabel: *ctrNamespaceLabel, CtrPodNameLabel: *ctrPodNameLabel, CtrNameLabel: *ctrNameLabel}
```

Comment:
Following previous comments on the fact that we should handle a list of metrics, here I would build that list based on the received flag. If the external-metrics for a given resourceType is not defined, the resource is not added into the list.

So instead of CpuMetrics, MemoryMetric you should have `map[ResourceName]string` that defines the metrics to be used for each resourceType.